extension/packagemanagerextension/nbextension/methods.js

```diff
@@ -0,0 +1,239 @@
+
+define([
+    'jquery',
+    './common',
+], function ($, common) {
```

Comment:
documentation, documentation, documentation

Reply:
This would seem better to be an "api" class (give it that name, including the filename), which converts all calls to the api into JS functions, ready to use (**and only api/ajax requests.. nothing more.. here you just return data**).
Also: you need to remove duplicated code (maybe one `ajax_request` function, with some configurations as params would suffice), remove urls hard coded all around the place (at least add a var on the top of the file to specify the base path to the server extension), etc. And try to be consistent on your coding style (example in the next comment block).

Reply:
Please try to replicate the style of SwanShare: https://gitlab.cern.ch/swan/jupyter/blob/qa/SwanShare/js/

Reply:
About the urls, they cannot be hardcoded ofc. They need to be relative and you need to allow possible base_paths, as necessary by JupyterHub, for example. But I expect you to know this because the exercise selection exercise you did had this detail and you solved it correctly.